File: OP_lab/lab5/c++/circle.cpp
//
// Created by maxim on 25.03.2020.
//

#include "circle.h"
#include <iostream>

circle::circle(Coord c, double r) : center{std::move(c)}, radius{r}
{}

double circle::perimeter() {
    return PI * 2.0 * radius;
}

double circle::area() {
    return PI * pow(radius, 2);
}

File: OP_lab/lab5/c#/Circle.cs
using System;

namespace lab_5_OP
{
    public class Circle : Figure
    {
        public Circle(double r)
        {
            _radius = r;
        }

        public override double Perimeter()
        {
            return Math.PI * 2.0 * _radius;
        }

        public override double Area()
        {
            return Math.PI * Math.Pow(_radius, 2);
        }

        private readonly double _radius;
    }
}